File: 2020/day_1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("day-1.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()
	var prev []int

	s := bufio.NewScanner(file)
scan:
	for s.Scan() {
		val, err := strconv.Atoi(s.Text())
		if err != nil {
			log.Fatal(err)
		}

		for _, p := range prev {
			if val+p == 2020 {
				fmt.Println(val * p)
				break scan
			}
		}

		prev = append(prev, val)
	}

	if err := s.Err(); err != nil {
		log.Fatal(err)
	}
}

File: 2020/day_1.py
prev = []
with open('day-1.txt') as fp:
    try:
        for line in fp:
            val = int(line.strip())
            for p in prev:
                if p + val == 2020:
                    print(p * val)
                    raise StopIteration
            prev.append(val)
    except StopIteration:
        pass
